src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -0,0 +1,139 @@
+    this.embeddings = changes.embeddings?.currentValue;
+    this.embeddingHandlesMap = changes.embeddingHandlesMap?.currentValue;
+    this.embeddingHandlesMap.forEach((value: any) => {
+      value.forEach((element: any) => {
+        element.isModelDownloaded = this.checkIfModelIsDownloaded(element.configString);
+      });
+    });
+  }
+
+  prepareDownloadedModels() {
+    [this.downloadedModelsQuery$, this.downloadedModelsList$] = this.projectApolloService.getModelProviderInfo();
+    this.subscriptions$.push(
+      this.downloadedModelsList$.subscribe((downloadedModels) => this.downloadedModels = downloadedModels));
+  }
+
+  checkForceHiddenHandles() {
+    const granularity = this.settingModals.embedding.create.embeddingCreationFormGroup.get('granularity').value;
+    const attId = this.settingModals.embedding.create.embeddingCreationFormGroup.get('targetAttribute').value;
+    const suggestionList = this.embeddingHandlesMap.get(attId)
+    for (let element of suggestionList) {
+      element = { ...element };
+      element.forceHidden = true;
+      if ((granularity == 'ON_ATTRIBUTE' && element.applicability?.attribute)
+        || (granularity == 'ON_TOKEN' && element.applicability?.token)) {
+        element.forceHidden = false;
+      }
+    }
+  }
+
+  deleteEmbedding() {
+    const embeddingId = this.settingModals.embedding.delete.id;
+    if (!embeddingId) return;
+    this.projectApolloService
+      .deleteEmbedding(this.project.id, embeddingId).pipe(first())
+      .subscribe(() => {
+        this.embeddings = this.embeddings.filter(e => e.id != embeddingId);
+        this.settingModals.embedding.create.blocked = !this.dataHandlerHelper.canCreateEmbedding(this.settingModals, this.embeddings);
+      });
+  }
+
+  addEmbedding() {
+    if (!this.dataHandlerHelper.canCreateEmbedding(this.settingModals, this.embeddings)) return;
+    const embeddingForm = this.settingModals.embedding.create.embeddingCreationFormGroup;
+    const embeddingHandle = embeddingForm.get("embeddingHandle").value;
+    const attributeId = embeddingForm.get("targetAttribute").value;
+    const granularity = embeddingForm.get("granularity").value;
+
+    this.projectApolloService.createEmbedding(this.project.id, attributeId, embeddingHandle, granularity.substring(3)).pipe(first()).subscribe();
+  }
+
+  selectFirstUnhiddenEmbeddingHandle(inputElement: HTMLInputElement) {
+    const suggestionList = this.embeddingHandlesMap.get(this.settingModals.embedding.create.embeddingCreationFormGroup.get("targetAttribute").value)
+    for (let embeddingHandle of suggestionList) {
+      if (!embeddingHandle.hidden && !embeddingHandle.forceHidden) {
+        this.selectEmbeddingHandle(embeddingHandle, inputElement);
+        return;
+      }
+    }
+  }
+
+  selectEmbeddingHandle(embeddingHandle: Embedding, inputElement: HTMLInputElement, hoverBox?: any) {
+    inputElement.value = embeddingHandle.configString;
+    hoverBox.style.display = 'none';
```

Comment:
seems to run into erros since hoverbox isn'T defined. Since this is an optional parameter I think we just need to check:
```suggestion
    if(hoverBox) hoverBox.style.display = 'none';
```

Reply:
also an issue in the current version of the app :D